layerindex/tools/github-fetch.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+# Github fetch utility
+
+# Copyright (C) 2017 Intel Corporation
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os
+import argparse
+import json
+import requests
+import re
+import subprocess
+
+def fetchall(args):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 24. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code blocks on lines 87-94, lines 96-106 into separate functions.